pdns/recursordist/test-syncres_cc10.cc

```diff
@@ -1563,7 +1563,7 @@ BOOST_AUTO_TEST_CASE(test_servestale_cname_to_nxdomain)
   BOOST_CHECK(ret[1].d_type == QType::A);
   BOOST_CHECK_EQUAL(ret[0].d_name, target);
   BOOST_CHECK_EQUAL(ret[1].d_name, DNSName("cname.powerdns.com"));
-  BOOST_CHECK_EQUAL(downCount, 4U);
+  BOOST_CHECK_EQUAL(downCount, 8U);
```

Comment:
I don't understand why we now seem to try to reach these servers twice?

Reply:
Oops, forgot about that, need to figure out the reason indeed.

Reply:
With this PR, we are seeing 8 timeouts. 4 for the `www.powerdns.com|A` query and 4 for the `cname.powerdns.com|A` query. 4 because we have two addresses (A and AAAA) and 2 NS.

With master, I do not see a query `cname.powerdns.com|A`, because the cache consultation stays in "serve-stale" mode (`d_serveStale` remains true for deeper recursions as well), so the cname record is immediately served from the cache with server-stale processing, while with the PR a resolve attempt is done before deciding to serve-stale. I think the second resolve is correct, as we should first try to get a fresh record before falling back to serving stale.

Reply:
Agreed, this makes more sense!